apps/ledger-live-mobile/src/components/RequiresBLE/hooks/useAndroidBLEPermissions.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  Permission,
+  PermissionsAndroid,
+  Platform,
+  AppState,
+} from "react-native";
+import { useRef, useCallback, useState, useEffect } from "react";
+
+const { PERMISSIONS, RESULTS } = PermissionsAndroid;
+export const bluetoothPermissions: Permission[] =
+  Platform.OS === "android"
+    ? Platform.Version < 31
+      ? []
+      : [PERMISSIONS.BLUETOOTH_SCAN, PERMISSIONS.BLUETOOTH_CONNECT]
+    : [];
+
+async function checkBluetoothPermissions() {
+  if (bluetoothPermissions.length === 0) return true;
+  return Promise.all(
+    bluetoothPermissions.map(permission =>
+      PermissionsAndroid.check(permission),
+    ),
+  ).then(results => results.every(res => res));
+}
+
+type RequestMultipleResult = {
+  allGranted: boolean;
+  generalStatus: typeof PermissionsAndroid.RESULTS[number];
+};
+
+async function requestBluetoothPermissions(): Promise<RequestMultipleResult> {
+  if (bluetoothPermissions.length === 0)
+    return { allGranted: true, generalStatus: RESULTS.GRANTED };
+  return PermissionsAndroid.requestMultiple(bluetoothPermissions).then(res => {
+    const permissionStatuses = Object.values(res);
+    const allGranted = permissionStatuses.every(
+      status => status === RESULTS.GRANTED,
+    );
+
+    /** https://developer.android.com/about/versions/11/privacy/permissions#dialog-visibility */
+    const someNeverAskAgain = permissionStatuses.some(
+      status => status === RESULTS.NEVER_ASK_AGAIN,
+    );
+
+    return {
+      allGranted,
+      generalStatus: allGranted
+        ? RESULTS.GRANTED
+        : someNeverAskAgain
+        ? RESULTS.NEVER_ASK_AGAIN
+        : RESULTS.DENIED,
+    };
+  });
+}
+
+const useAndroidBLEPermissions = () => {
```

Comment:
Should we put in the same location those new hooks together with `usePromptBluetoothCallback` ?